core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -2460,8 +2449,25 @@ public int getTenantId() throws UserStoreException {
         String userPropertyName =
                 realmConfig.getUserStoreProperty(LDAPConstants.USER_NAME_ATTRIBUTE);
 
-        searchFilter = "(&" + searchFilter + "(" + property + "=" + escapeSpecialCharactersForFilterWithStarAsRegex(
-                value) + "))";
+        if ("objectGUID".equals(property)) {
```

Comment:
"objectGUID" is used in multiple places in the code. Therefore we can put this in a private constant within the class and use it.